proposals/hasCrossSiteAncestor.md

```diff
@@ -0,0 +1,116 @@
+# Proposal: Update Extensions to interact with Partitioned Cookies containing Cross-Site Ancestor Chain Bit values
+
+**Summary**
+
+Allow extensions to utilize a hasCrossSiteAncestor boolean value when interacting with partitioned cookies that include a cross-site ancestor chain bit in their partiton key.
+
+#### Background information: Description of Cross-site Ancestor Chain Bit in partitioned cookies
+
+The cross-site ancestor chain bit is a component of the cookie partition key that is set by the browser when keys are created. If the bit indicates true, it means the cookie has been set in a third-party context.
+
+A third-party context occurs when any parent frame, in the frame tree, is cross-site with the frame or subresource the cookie is being set on. If one site of the frame tree is third-party, then all children frames of that frame are also third-party and have a cross site ancestor. Third-party contexts also occur when the top-level frame makes a third-party subresource request and when a first-party request is redirected to a thrid-party URL (even if it is subsequently redirected back to a first-party request after, that subsequent first-party request is now considered an ABA request).
```

Comment:
Due to community [feedback](https://github.com/privacycg/CHIPS/issues/40#issuecomment-2178846023) and compatibility issues, a redirect and opener are not being considered cross-site in the chromium implementation. I'm fine with the language @Rob--W, proposed as it is correct but I wanted to get some opinion on if it makes sense to alter the language to reflect the implementation or keep it as is. 

@oliverdunk, @DCtheTall

Reply:
Is the term "redirect entry" defined somewhere? Before the Chromium implementation incorporated the [feedback](https://github.com/privacycg/CHIPS/issues/40#issuecomment-2178846023), HTTP redirects behaved differently than Javascript redirects. Would a Javascript redirect not count as a "redirect entry"?

Reply:
> A third-party context occurs when the subresource the cookie is being set on has ...

How is this applied to a cross-site fetch request whose response sets a partitioned cookie? What's the "subresource" in this case?

Reply:
@HeikoTheissen, in the case of extensions, there isn't a subresource. 

Since the ancestor chain bit is a new element of the cookie partition key, this language was added to provide developers, who don't have prior knowledge of the ancestor chain bit, a definition to help understand it's purpose.

Reply:
> Due to community [feedback](https://github.com/privacycg/CHIPS/issues/40#issuecomment-2178846023) and compatibility issues, a redirect and opener are not being considered cross-site in the chromium implementation. I'm fine with the language @Rob--W, proposed as it is correct but I wanted to get some opinion on if it makes sense to alter the language to reflect the implementation or keep it as is.

Adjust the phrasing to match the implementation / final design. The purpose of my suggested language was to clarify the actual behavior, and that should closely match the final design. If there are parts that may change, it would also be fine to call out that the behavior may change as the implementation evolves and the spec matures.

If you are in favor of the suggested text, providing review feedback may be easier if you apply the suggested text. Then other reviewers can suggest new improvements to the (applied) suggestion instead of trying to somehow read the diff from my comment.

> > > A third-party context occurs when the subresource the cookie is being set on has ...
> >
> > How is this applied to a cross-site fetch request whose response sets a partitioned cookie? What's the "subresource" in this case?
>
> @HeikoTheissen, in the case of extensions, there isn't a subresource.

This response puzzles me. By the term "subresource", I meant all non-document requests. What meaning of "subresource" are you using? Could you elaborate what you mean by "in the case of extensions, there isn't a subresource"?

Reply:
> By the term "subresource", I meant all non-document requests.

That makes sense. (I had initially thought of the subresource as a frame, image or something in the document tree.)

Reply:
Thank you for the suggestion, I've applied the text.

I should have been more specific with my language as upon reading it again it's clear I misrepresented my intent. 

In the context of the issue that, @HeikoTheissen, we had been discussing redirect requests from a subresource and how the browser calculates the value of the ACB. The issue that was encountered, wouldn't impact this API not because there is no subresource but because the API has the ability to set the value of the ACB directly which webpages do not.